frontend/src/pages/profile/Profile.tsx

```diff
@@ -10,24 +10,87 @@ import {
   ProfileHeader,
   ProfileNotes,
   Loading,
+  Button,
 } from '@/components';
-import { Status, Role, Program } from '@/common';
+import { useState } from 'react';
+import { Status, Role, Program, ButtonTypes } from '@/common';
 import { useProgramFieldData } from '@/hooks/useProgramFieldData';
 import { ProfileBreadcrumbs, ProfileToggle } from '@/components/profile/common';
 import { useRoleContext } from '@/providers';
 import { RecommitmentDetails } from '@/components/profile/details/RecommitmentDetails';
 import { useRecommitmentCycle } from '@/hooks/useRecommitment';
+import { RecommitmentStatus } from '../../common/enums/recommitment-status';
+import { RecommitmentConfirmation } from './RecommitmentConfirmation';
 
 const Profile = () => {
-  const { loading, roles, personnel, updatePersonnel, profileData } = usePersonnel();
+  const { loading, roles, personnel, updatePersonnel, profileData, fetch } =
+    usePersonnel();
   const { program } = useRoleContext();
   const { functions, bcwsRoles } = useProgramFieldData(program);
-  const {isRecommitmentCycleOpen} = useRecommitmentCycle();
+  const {
+    isRecommitmentCycleOpen,
+    isRecommitmentReinitiationOpen,
+    updateRecommitment,
+  } = useRecommitmentCycle();
+  const [confirmReinitiateOpen, setConfirmReinitiateOpen] = useState(false);
 
   if (loading) {
     return <Loading />;
   }
 
+  const recommitmentStatusText = () => {
+    const recommitment = personnel?.recommitment?.find(
```

Comment:
Looks good - tiny nit - I was wondering if we might consider moving these into the useRecommitment hook? Just for separation of concerns?

Reply:
Good idea